src/peft/utils/other.py

```diff
@@ -560,12 +561,18 @@ def set_adapter(self, adapter_names: Union[str, list[str]]):
         if len(adapter_names) > 1:
             raise ValueError(f"Attempted to set multiple ({adapter_names}) adapters at once for modules_to_save.")
 
+        if len(adapter_names) == 0:
+            # when calling model.add_adapter, the new adapter is not automatically active
+            self._active_adapter = []
+            return
+
         adapter_name = adapter_names[0]
 
         if adapter_name not in self._adapters:
             raise ValueError(f"Adapter {adapter_name} not found in {self._adapters}")
 
-        self.modules_to_save[self.active_adapters[0]].requires_grad_(False)
+        for currently_active_adapter_name in self.active_adapters:
+            self.modules_to_save[currently_active_adapter_name].requires_grad_(False)
```

Comment:
In case of `adapter_names == []` this will not reset the `requires_grad` field. Wouldn't it be best to do it similar to `enable_adapters` and use `self.modules_to_save.requires_grad_(False)` and then enable the adapters that should be enabled?